### PR TITLE
When connecting to a Postgres database through a datalink stored on Enso Cloud, its asset ID is included in the audit logs

### LINKED ISSUE #9869: Correlate Database audit log entries with the asset id of the data link used for the connection, if available
If a database connection was opened through a data link, it is audited. If the data link was from the Enso Cloud, we want to add its `assetId` to the metadata of each log entry.

For data links that are _not_ in the Cloud (local file system etc.) we cannot have a stable identifier like the `assetId`, so we don't add anything similar.

- [x] The data link _instance_ fetched from the Cloud needs to be aware of its asset id - modify the data link read logic to be able to supply optional id.
- [x] That ID needs to be passed to the `CloudAuditedConnection`, probably through additional JDBC properties.
- [ ] Tests need to be written
	- The challenge is: our Cloud mock does not support reading files/datalinks, but we wanted to rely on the mock for testing the logs. Some solution needs to be picked:
	  1. Add basic datalink support to the mock, and run the tests on there.
	  2. Run the tests on the real cloud deployment (in local testing the stability seemed okayish - but hard to tell how it will be on CI).
	     - the benefit is we get a real integration test, as the mock and cloud can diverge,
	     - we still need to implement #9673 first, to be able to prepare the data link used in the test.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #9869

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_Data_Link.enso [hunk 1]
```diff
@@ -19,7 +19,7 @@ type S3_Data_Link
     Value (uri : Text) format_json (credentials : AWS_Credential)
 
     ## PRIVATE
-    parse json -> S3_Data_Link =
+    parse json _ -> S3_Data_Link =
```
Comment (jdunkerley) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_Data_Link.enso]: I know we are discarding this but can we do:
```suggestion
    parse json source -> S3_Data_Link =
        _ = source
```
as clearer to read.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Enso_File_Data_Link.enso [hunk 2]
```diff
@@ -21,7 +21,7 @@ type Enso_File_Data_Link
     private Value (path : Text) format_json
 
     ## PRIVATE
-    parse json -> Enso_File_Data_Link =
+    parse json _ -> Enso_File_Data_Link =
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Enso_File_Data_Link.enso]: as above.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Internal/HTTP_Fetch_Data_Link.enso [hunk 3]
```diff
@@ -20,7 +20,7 @@ type HTTP_Fetch_Data_Link
     Value (request : Request) format_json
 
     ## PRIVATE
-    parse json -> HTTP_Fetch_Data_Link =
+    parse json _ -> HTTP_Fetch_Data_Link =
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Network/HTTP/Internal/HTTP_Fetch_Data_Link.enso]: as above

File: distribution/lib/Standard/Snowflake/0.0.0-dev/src/Snowflake_Data_Link.enso [hunk 4]
```diff
@@ -19,7 +19,7 @@ type Snowflake_Data_Link
     Table name:Text details:Snowflake_Details
 
     ## PRIVATE
-    parse json -> Snowflake_Data_Link =
+    parse json _ -> Snowflake_Data_Link =
```
Comment (jdunkerley) [inline distribution/lib/Standard/Snowflake/0.0.0-dev/src/Snowflake_Data_Link.enso]: last one!

**Review by jdunkerley — APPROVED** [review]
Looks good - small nit from my styling preferences

File: distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_Data_Link.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_Data_Link.enso]: Ok sure, I was also not sure if `_` is right here, I think that named argument is indeed better - will amend all of them

**Review by AdRiley — APPROVED** [review]

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**